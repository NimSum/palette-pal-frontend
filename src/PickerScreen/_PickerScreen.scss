.PickerScreen {
	display: flex;
	flex-direction: column;
	min-height: calc(100vh - 102px);
}

.palette-display {
	flex: auto;
	display: flex;
	width: 100%;

	.color {
		flex: 1;
		display: flex;
		flex-direction: column;
		justify-content: center;
		position: relative;
	}

	.color1 {
		background: #214046;
	}

	.color2 {
		background: #05596a;
	}

	.color3 {
		background: #6bcba5;
	}

	.color4 {
		background: #9ee6a9;
	}

	.color5 {
		background: #cbf3c3;
	}

	.color-value {
		font-size: 20px;
		color: white;
		opacity: .3;
	}

	.hold {
		font-size: 26px;
    color: white;
    font-weight: 600;
		opacity: .4;
		position: absolute;
		bottom: 30px;
		left: 0;
		right: 0;
		margin-left: auto;
		margin-right: auto;
	}
}

.picker-footer {
	background: white;
	position: absolute;
	bottom: 0;
	height: 130px;
	width: 100vw;
	display: flex;
	padding: 30px;
	align-items: center;
	justify-content: space-between;

	.instructions {
		font-size: 20px;
		color: $grey-text;
	}

	.save-btn {
		background: $btn-color;
		font-size: 20px;
		color: white;
		padding: 10px 40px;
		border-radius: 5px;

		&:hover {
			background: $btn-hover;
			cursor: pointer;
		}
	}
}

.fa,
.fas,
.far {
	margin-right: 8px;
}
