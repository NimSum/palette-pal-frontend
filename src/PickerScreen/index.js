import React, { Component } from 'react';
import SubHeader from '../SubHeader';

class PickerScreen extends Component {

  render() {
    return (
      <>
        <SubHeader />
        <section className="PickerScreen">
          <div className="palette-display">
            <div className="color color1">
              <p className="color-value">#214046</p>
              <p className="hold"><i className="far fa-check-square"></i>HOLD</p>
            </div>
            <div className="color color2">

            </div>
            <div className="color color3">

            </div>
            <div className="color color4">

            </div>
            <div className="color color5">

            </div>
          </div>
          <div className="picker-footer">
            <p className="instructions"><i className="fas fa-sync-alt" aria-hidden="true"></i>Press <strong>space</strong> to refresh unselected colors</p>
            <button className="save-btn">
              <i className="far fa-save" aria-hidden="true"></i>Save Palette
            </button>
          </div>
        </section>
      </>
    );
  }
}

export default PickerScreen;