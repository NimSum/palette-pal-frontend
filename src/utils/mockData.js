const mockData = {
  mockDetailedProjects: [
    {
      "project_name": "Project ONE",
      "project_id": 1,
      "palette_name": "Palette ONE",
      "palette_id": 1,
      "color_1": "#000000",
      "color_2": "#000000",
      "color_3": "#319547",
      "color_4": "#000000",
      "color_5": "#834267"
    },
    {
      "project_name": "Project TWO",
      "project_id": 2,
      "palette_name": "Palette TWO",
      "palette_id": 2,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267"
    },
    {
      "project_name": "Project ONE",
      "project_id": 2,
      "palette_name": "Palette THREE",
      "palette_id": 3,
      "color_1": "#123456",
      "color_2": "#654321",
      "color_3": "#245216",
      "color_4": "#784895",
      "color_5": "#235686"
    }
  ],
  mockProjects: [
    {
      "id": 1,
      "project_name": "Project ONE",
      "created_at": "2019-07-05T21:39:51.436Z",
      "updated_at": "2019-07-05T21:39:51.436Z"
    },
    {
      "id": 2,
      "project_name": "Project TWO",
      "created_at": "2019-07-05T21:40:51.436Z",
      "updated_at": "2019-07-05T21:40:51.436Z"
    }
  ],
  mockPalettes: [
    {
      "id": 1,
      "palette_name": "Test ONE",
      "project_id": 1,
      "color_1": "#010101",
      "color_2": "#925578",
      "color_3": "#319547",
      "color_4": "#796876",
      "color_5": "#834267",
      "created_at": "2019-07-05T21:39:57.227Z",
      "updated_at": "2019-07-05T21:39:57.227Z"
    },
    {
        "id": 2,
        "palette_name": "Test TWO",
        "project_id": 2,
        "color_1": "#000000",
        "color_2": "#000000",
        "color_3": "#319547",
        "color_4": "#000000",
        "color_5": "#834267",
        "created_at": "2019-07-06T01:24:45.881Z",
        "updated_at": "2019-07-06T01:24:45.881Z"
    }
  ]
}

export default mockData;