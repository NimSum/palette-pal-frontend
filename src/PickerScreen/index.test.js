import React from 'react';
import PickerScreen from './index';
import { shallow } from 'enzyme';
import mockData from '../utils/mockData';
jest.mock('../utils/colorFormatter');

describe('PickerScreen', () => {
  let wrapper;
  let instance;
  const mockUpdatePaletteData = jest.fn();
  const mockUpdateProjectData = jest.fn();
  const mockShowAcctDialog = jest.fn();
  window.addEventListener = jest.fn();
  window.removeEventListener = jest.fn();

  beforeEach(() => {
    wrapper = shallow(< PickerScreen
      data={[1, 2, 3]}
      updateProjectData={mockUpdateProjectData}
      updatePaletteData={mockUpdatePaletteData}
      showAcctDialog={mockShowAcctDialog}
      />,
      { disableLifecycleMethods: true });
    
    instance = wrapper.instance();

    jest.spyOn(instance, 'generateNewPalette');
    jest.spyOn(instance, 'toggleHold');
    jest.spyOn(instance, 'refreshUnheldColors');
    jest.spyOn(instance, 'closeDialog');
    jest.spyOn(instance, 'setPickerOption');
    jest.spyOn(instance, 'updateColor');
  })
  
  it('should match component PickerScreen snapshot', () => {
    wrapper.setState({ colors: mockData.justColors })
    expect(wrapper).toMatchSnapshot();
  });

  it('should have the expected default state', () => {
    expect(wrapper.state()).toEqual({
      colors: {
        color_1: '#fff',
        color_2: '#fff',
        color_3: '#fff',
        color_4: '#fff',
        color_5: '#fff'
      },
      held: [],
      showSaveDialog: false,
      format: 'hex',
      mode: 'random'
    })
  })

  it('should call generatePalette and add a window event listener on mount', () => {
    instance.componentDidMount();

    expect(instance.generateNewPalette).toHaveBeenCalled();

    expect(window.addEventListener).toHaveBeenCalled();
  })

  it('should call remove the window event listener when component unmounts', () => {
    instance.componentWillUnmount();

    expect(window.removeEventListener).toHaveBeenCalled();
  })

  it('should get a random hex code when getRandomColor is called', () => {
    const result = instance.getRandomColor();

    expect(typeof result).toEqual('string');
    expect(result.length).toEqual(7);
  })

  it('should update colors in state when getRandomPalette is called', () => {
    expect(wrapper.state('colors')).toEqual({
      color_1: '#fff',
      color_2: '#fff',
      color_3: '#fff',
      color_4: '#fff',
      color_5: '#fff'
    });

    instance.generateNewPalette();

    expect(wrapper.state('colors.color_1')).not.toBe('#fff');
  })

  it('should return an object with five new colors from state when getRandomPalette is called', () => {
    expect(wrapper.state('colors').color_1).toEqual('#fff');

    const result = instance.getRandomPalette();

    expect(result.color_1).not.toBe('#fff');
    expect(typeof result.color_1).toEqual('string');
    expect(result.color_1.length).toEqual(7);
  })

  it('should add or remove color to "held" in state when toggleHold is called based on whether it is already in there or not', () => {
    expect(wrapper.state('held')).toEqual([]);

    instance.toggleHold('#456542');

    expect(wrapper.state('held')).toEqual(['#456542']);

    instance.toggleHold('#778877');

    expect(wrapper.state('held')).toEqual(['#456542', '#778877']);

    instance.toggleHold('#456542');

    expect(wrapper.state('held')).toEqual(['#778877']);
  })

  it('should update colors not in "held" when refreshUnheldColors is called', () => {
    const mockEvent = { keyCode: 32 };
    instance.generateNewPalette();
    const color1 = wrapper.state('colors').color_1;
    const color2 = wrapper.state('colors').color_2;
    instance.toggleHold('color_1');

    expect(wrapper.state('held')).toEqual(['color_1']);

    instance.refreshUnheldColors(mockEvent);

    expect(wrapper.state('colors').color_1).toEqual(color1);
    expect(wrapper.state('colors').color_2).not.toBe(color2);
  })

  it('should update showSaveDialog in state to true when the save palette button is clicked and the user is logged in', () => {
    expect(wrapper.state('showSaveDialog')).toEqual(false);

    wrapper.find('.save-btn').simulate('click');

    expect(wrapper.state('showSaveDialog')).toEqual(true);
  })

  it('should update showSaveDialog in state to false when closeDialog is called', () => {
    wrapper.find('.save-btn').simulate('click');
    expect(wrapper.state('showSaveDialog')).toEqual(true);

    instance.closeDialog();

    expect(wrapper.state('showSaveDialog')).toEqual(false);
  })

  it('should update an option in state and call generateNewPalette if the option is mode when setPickerOption is called', () => {
    expect(wrapper.state('mode')).toEqual('random');
    instance.setPickerOption({ mode: 'test' });

    expect(wrapper.state('mode')).toEqual('test');
    expect(instance.generateNewPalette).toHaveBeenCalled();
  })

  it('should update a color in state when updateColor is called', () => {
    expect(wrapper.state('colors').color_1).toEqual('#fff');

    instance.updateColor('#000000', 'color_1');

    expect(wrapper.state('colors').color_1).toEqual('#000000');
  })
  
})