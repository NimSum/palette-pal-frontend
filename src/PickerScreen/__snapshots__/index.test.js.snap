// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickerScreen shoud match component PickerScreen snapshot 1`] = `ShallowWrapper {}`;
